samples/matter/lock/README.rst

```diff
@@ -78,25 +78,24 @@ For details, see the `Commissioning the device`_ section.
 Thread and Wi-Fi switching
 ==========================
 
-When built using the ``thread_wifi_switched`` build type and programmed to the nRF5340 DK with the nRF7002 EK shield attached, the door lock sample supports a feature that allows you to :ref:`dynamically switch between Matter over Thread and Matter over Wi-Fi <ug_matter_overview_architecture_integration_designs_switchable>` on the device boot.
-Due to internal flash size limitations, only one transport protocol can be used at a time.
+When built using the ``thread_wifi_switched`` build type and programmed to the nRF5340 DK with the nRF7002 EK shield attached, the door lock sample supports a feature that allows you to :ref:`switch between Matter over Thread and Matter over Wi-Fi <ug_matter_overview_architecture_integration_designs_switchable>` at runtime.
+Due to Matter protocol limitations, a single Matter node can only use one transport protocol at a time.
 
 .. matter_door_lock_sample_thread_wifi_switch_desc_start
 
-The application is built in two variants: the first one is the application working with Matter over Thread and the second one is the application working with Matter over Wi-Fi.
-You can configure which transport is selected on the device boot as the default one.
-Both application variants are programmed into separate partitions of the external flash.
-The application runs from the internal flash memory, using one of the variants from the external flash.
-You can trigger the switch from one variant to another using the **Button 3** on the nRF5340 DK.
-The device is rebooted into the MCUboot bootloader, which replaces the current variant by swapping the application variant in the internal flash.
+The application is built with support for both Matter over Thread and Matter over Wi-Fi.
+The device activates either Thread or Wi-Fi transport protocol on boot, based on a flag stored in the non-volatile memory on the device.
+By default, Matter over Wi-Fi is activated.
 
-.. note::
-   Because the external flash is used for both the DFU and the switching feature, this implementation has higher memory size requirements and you need an external flash with at least 6 MB of memory.
+You can trigger the switch from one transport protocol to the other using the **Button 3** on the nRF5340 DK.
+This toggles the flag stored in the non-volatile memory, and then the device is factory reset and rebooted.
+The factory reset and recommissioning to a Matter fabric allows the device to be provisioned with network credentials for the transport protocol that it was switched to, and to start operating in the selected network.
+
+The factory reset does not switch the device back to the default transport protocol (Wi-Fi).
```

Comment:
```suggestion
This toggles the flag stored in the non-volatile memory, and then the device is factory reset and rebooted.
Because the flag is toggled, the factory reset does not switch the device back to the default transport protocol (Wi-Fi).
Instead, the factory reset and recommissioning to a Matter fabric allows the device to be provisioned with network credentials for the transport protocol that it was switched to, and to start operating in the selected network.
```
Reorganized this one a bit.